modules/templates/handler.rb

```diff
@@ -7,23 +7,29 @@ module Proxy::Templates
   class Handler < ::Proxy::HttpRequest::ForemanRequest
     include Proxy::Log
 
-    def get_template(kind, token, static = false)
-      opts = { :token => token,
-               :url   => Proxy::Templates::Plugin.settings.template_url
-      }
-      opts[:static] = static if static
-      request = request_factory.create_get("/unattended/#{kind}", opts)
-      logger.debug "retrieving a template from %s/%s" % [Proxy::Templates::Plugin.settings.template_url, request.path]
-      res = send_request(request)
+    def get_template(kind, request, params, headers)
+      url = Proxy::Templates::Plugin.settings.template_url || raise("Setting template_url not set!")
+      proxy_ip = URI.parse(url).host rescue "127.0.0.1"
```

Comment:
Instead of this and the line above, set a default value for `template_url` setting in https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/templates_plugin.rb